tokio/src/signal/unix.rs

```diff
@@ -325,7 +325,9 @@ fn signal_enable(signal: c_int, handle: Handle) -> io::Result<()> {
 #[must_use = "streams do nothing unless polled"]
 #[derive(Debug)]
 pub struct Signal {
+    signal: c_int,
     rx: Receiver<()>,
+    poll_rx: Option<Pin<Box<Recv<'static, ()>>>>,
 }
```

Comment:
Another option here is to copy the code from [`BroadcastStream`][1] somewhere into a private module and use its poll method. Yet another option would be to write a self-referential struct, which can be done with a box.

[1]: https://docs.rs/tokio-stream/0.1/tokio_stream/wrappers/struct.BroadcastStream.html

Reply:
Thanks for the tip, I'll take a closer look